test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -382,16 +382,78 @@ var _ = ginkgo.Describe("end to end testing", func() {
 			util.ValidateNumDeployments(ctx, wec1, ns, 1)
 			util.ValidateNumDeployments(ctx, wec2, ns, 0)
 			util.ValidateSingletonStatus(ctx, wds, ns, "nginx-singleton")
-			patch_again := []byte(`{"spec":{"clusterSelectors":[{"matchLabels":{"name":"CelestialNexus"}}]}}`)
+			patchAgain := []byte(`{"spec":{"clusterSelectors":[{"matchLabels":{"name":"CelestialNexus"}}]}}`)
 			_, err = ksWds.ControlV1alpha1().BindingPolicies().Patch(
-				ctx, "nginx-singleton", types.MergePatchType, patch_again, metav1.PatchOptions{})
+				ctx, "nginx-singleton", types.MergePatchType, patchAgain, metav1.PatchOptions{})
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 			util.ValidateNumDeployments(ctx, wec1, ns, 0)
 			util.ValidateNumDeployments(ctx, wec2, ns, 0)
 			util.ValidateSingletonStatusZeroValue(ctx, wds, ns, "nginx-singleton")
 		})
 	})
 
+	ginkgo.Context("singleton status eventual consistency", func() {
+		ginkgo.It("cleans up previously synced but currently invalid singleton status", func(ctx context.Context) {
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.CreateDeployment(ctx, wds, ns, "nginx-singleton",
+				map[string]string{
+					"app.kubernetes.io/name": "nginx-singleton",
+				})
+			util.CreateBindingPolicy(ctx, ksWds, "nginx-singleton",
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"name": "cluster1"}},
+				},
+				[]ksapi.DownsyncPolicyClause{
+					{DownsyncObjectTest: ksapi.DownsyncObjectTest{
+						ObjectSelectors: []metav1.LabelSelector{{MatchLabels: map[string]string{"app.kubernetes.io/name": "nginx-singleton"}}},
+					}},
+				},
+			)
+			BPPatch := []byte(`{"spec":{"wantSingletonReportedState": true}}`)
+			_, err := ksWds.ControlV1alpha1().BindingPolicies().Patch(
```

Comment:
FYI, It would be OK with me if the BindingPolicy were created with the Spec that this test needs.

Reply:
I see that func `CreateBindingPolicy` currently doesn't have an argument to pass the needed value of `wantSingletonReportedState`, so `Patch`es are used whenever that value needs to be set.

Are you suggesting some change to `CreateBindingPolicy`?

Reply:
There are a number of possible ways to accomplish that.

You could generalize CreateBindingPolicy by adding a parameter specifically for that. Or by adding a variadic parameter (so that the existing calls do not have to be edited) taking any sequence of functions that tweak the constructed BindingPolicy. Or recognize that the Get check in CreateBindingPolicy is pointless (the success of the Create implies that Get will succeed) and just construct the BindingPolicy you want and call Create directly and expect no error.

Reply:
Adopted the variadic parameter approach.